kernel/arch/dreamcast/hardware/g1ata.c

```diff
@@ -221,15 +220,21 @@ inline int g1_ata_mutex_unlock(void) {
 
 static void g1_ata_set_sector_and_count(uint64_t sector, uint32_t count, int lba28) {
     if(!lba28) {
+        if(count == 65536) {
+            count = 0;
+        }
         OUT8(G1_ATA_SECTOR_COUNT, (uint8_t)(count >> 8));
         OUT8(G1_ATA_LBA_LOW, (uint8_t)((sector >> 24) & 0xFF));
         OUT8(G1_ATA_LBA_MID, (uint8_t)((sector >> 32) & 0xFF));
         OUT8(G1_ATA_LBA_HIGH, (uint8_t)((sector >> 40) & 0xFF));
     }
+    else if(count == 256) {
+        count = 0;
+    }
```

Comment:
Same here, you can remove this because `(uint8_t)count` for count == 256 and also for count == 65536 will resolve to 0.